src/Downloader.ts

```diff
@@ -613,7 +643,7 @@ class Downloader {
   private async downloadImage(url: string, handler: any) {
     try {
       this.s3.downloadBlob(stripHttpFromUrl(url)).then(async (s3Resp) => {
-        if (s3Resp?.Metadata?.etag) {
+        if (s3Resp?.Metadata?.etag && s3Resp.Metadata.webp === (!this.cssDependenceUrls.hasOwnProperty(url) && this.webp ? 'webp' : '1')) {
```

Comment:
`&& s3Resp.Metadata.webp === (!this.cssDependenceUrls.hasOwnProperty(url) && this.webp ? 'webp' : '1'` should be removed. Within a run, we write everything with the same version tag.